### DISCUSSION TITLE
Insert and Update statement with same values.

### DISCUSSION THREAD

**Discussion by nipunpatel86** [nhibernate/nhibernate-core#3143] [Q&A]
First during entity save it creates insert statements and than on commit it creates update statement.The value during insert statement is same in update statement and it's foreign key.There are 600+ update statements executed during transaction so it will impact on performance.
How to restricts update statement as it is not update any values?

**Comment by fredericDelaporte** [comment]
Can you please provide test code demonstrating this behavior? See [here](https://github.com/nhibernate/nhibernate-core/blob/master/CONTRIBUTING.md#creating-a-test-case-to-verify-the-issue).
